### PR TITLE
Fix broken "How to reproduce" lines in CSV tests

### LINKED ISSUE #105017: ESQL: broken repro lines in csv tests
CSV test cases with a [skip annotation](https://github.com/elastic/elasticsearch/blob/7ce8d76559f89e39b17ba1ab713b7370f37454fb/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec#L949) cannot be run individually, and when they fail the reproduction line does not work (`No tests found for given includes`).

To reproduce, run the csv tests but change expected result for [groupByNull](https://github.com/elastic/elasticsearch/blob/7ce8d76559f89e39b17ba1ab713b7370f37454fb/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec#L954) and attempt to use the repro line.

The issue seems to be caused by the `+` sign used in the skip reason. If we change the csv test header from
```
groupByNull#[skip:-8.12.99,reason:bug fixed in 8.13+]
```
to
```
groupByNull#[skip:-8.12.99,reason:bug fixed in 8.13]
```
the repro line works as expected.

### PR DESCRIPTION
**Pull request by ivancea**
Fixes https://github.com/elastic/elasticsearch/issues/105017

A quick summary:
- `randomizedtesting` library doesn't handle "+" chars in test names
- We currently have the "...#[skip:..., ...]" instructions as part of the test name

As removing the instructions from the name fixes both issues (And makes launching a single test in CLI far easier), the instructions have been moved out of the test name to another parameter.

Before:
```
gradlew ':x-pack:plugin:esql:test' --tests "org.elasticsearch.xpack.esql.CsvTests" -Dtests.method="test {stats.GroupByNull#[skip:-8.12.99,reason:bug fixed in 8.13+]}"
```

After:
```
gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.CsvTests" -Dtests.method="test {stats.GroupByNull}"
```

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ public void testFailure(Failure failure) throws Exception {
         boolean isBwcTest = Boolean.parseBoolean(System.getProperty("tests.bwc", "false"));
 
         // append Gradle test runner test filter string
-        b.append("'" + task + "'");
+        b.append("\"" + task + "\"");
```
Comment (ivancea) [inline test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java:70]: Single quotes doesn't work in Windows. Double quotes should work everywhere

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java
Comment (ivancea) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java:1]: I'm not sure if this is worth it porting to the sql module, that has the same classes and logics. Looking for opinions here

File: test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java (hunk 1 above)
Comment (nik9000) [reply test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java:70]: I think we went single quotes here so we don't accidentally do glob pattern matching or something. At least, that's why I'd have used single quotes if it were me.

**Review by nik9000 — COMMENTED** [review]
I'll pull it locally to try just out of paranoia, but seems fine.

**Review by nik9000 — APPROVED** [review]

File: test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java (hunk 1 above)
Comment (ivancea) [reply test/framework/src/main/java/org/elasticsearch/test/junit/listeners/ReproduceInfoPrinter.java:70]: Looking into this, I reached the conclusion that this "shouldn't" affect gradle, as this is the CLI escaping mechanism? Not sure, and not an expert. But as it works in Windows, Mac, and Linux (from your other comment)... This is just a error message anyway :dancing-blob:

**Pull request merged**